src/sock.c

```diff
@@ -149,11 +161,53 @@ int sock_set_keepalive(sock_t sock, int timeout, int interval)
             return ret;
 #endif /* TCP_KEEPINTVL */
     }
+
+    if (count) {
+#ifdef TCP_KEEPCNT
+        ret = setsockopt(sock, IPPROTO_TCP, TCP_KEEPCNT, &count, sizeof(count));
+        if (ret < 0)
+            return ret;
+#endif /* TCP_KEEPCNT */
+    }
+
+    if (user_timeout) {
+#ifdef TCP_USER_TIMEOUT
+        ret = setsockopt(sock, IPPROTO_TCP, TCP_USER_TIMEOUT, &user_timeout,
+                         sizeof(user_timeout));
+        if (ret < 0)
+            return ret;
+#elif defined(TCP_RXT_CONNDROPTIME)
+        int rxt = user_timeout / 1000;
+        ret = setsockopt(sock, IPPROTO_TCP, TCP_RXT_CONNDROPTIME, &rxt,
+                         sizeof(rxt));
+        if (ret < 0)
+            return ret;
+#endif /* TCP_USER_TIMEOUT */
+    }
+
 #endif /* _WIN32 */
 
     return ret;
 }
 
+/** Example sockopt callback function
+ *  An example function that can be used to set reasonable default keepalive
+ *  options on sockets when registered for a connection with
+ *  xmpp_conn_set_sockopt_callback()
+ *
```

Comment:
```suggestion
 *
 *  @param conn a Strophe connection object
 *  @param socket pointer to a socket descriptor
 *
 *  @see xmpp_sockopt_callback for details on the `socket` parameter
```

just for the sake of completeness :)